File: data/multilang_comparison/program_go/src/func40.go
package main
import "math/rand"
var _ = rand.Intn
func Func40(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array10 *Array
   if pCounter > 0 {
       array10 = vars.Data[pCounter-1]
       pCounter--
       array10.RefC++
   } else {
       array10 = &Array{
           Size: 42,
           RefC: 1,
           Id: 10,
           Data: make([]uint32, 42),
       }
   }
   for i := 0; i < array10.Size; i++ {
       array10.Data[i]--
   }
   for i := 0; i < array10.Size; i++ {
       if array10.Data[i] == 21 {
           return array10
       }
   }
   return array10;
}

File: data/multilang_comparison/program_cpp_array/src/func40.cpp
#include "program_cpp_array.hpp" 
Array* func40(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array10;
   if (pCounter > 0) {
      array10 = vars->data[--pCounter];
      array10->refC++;
      DEBUG_COPY(array10->id);
   } else {
      array10 = new Array();
      array10->size = 42;
      array10->refC = 1;
      array10->id = 10;
      array10->data = new unsigned int[array10->size]();
      DEBUG_NEW(array10->id);
   }
   for (int i = 0; i < array10->size; i++) {
       array10->data[i]--;
   }
   for (int i = 0; i < array10->size; i++) {
       if (array10->data[i] == 21) {
           return array10;
       }
   }
   return array10;
}

